server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexAliasesService.java

```diff
@@ -244,4 +238,52 @@ private static void validateAliasTargetIsNotDSBackingIndex(ClusterState currentS
             );
         }
     }
+
+    /**
+     * A cluster state update task that applies the alias actions to the given cluster state.
+     */
+    private class ApplyAliasActions implements ClusterStateTaskListener {
+
+        private final Iterable<AliasAction> actions;
+        private final ActionListener<AcknowledgedResponse> listener;
+
+        ApplyAliasActions(Iterable<AliasAction> actions, ActionListener<AcknowledgedResponse> listener) {
+            this.actions = actions;
+            this.listener = listener;
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            listener.onFailure(e);
+        }
+
+        private void onSuccess() {
+            listener.onResponse(AcknowledgedResponse.of(true));
+        }
+
+        /**
+         * Handles the cluster state transition to a version that reflects the provided {@link AliasAction}s by calling
+         * the {@link MetadataIndexAliasesService#applyAliasActions(ClusterState, Iterable)}.
+         */
+        public ClusterState execute(ClusterState currentState) {
+            return MetadataIndexAliasesService.this.applyAliasActions(currentState, actions);
```

Comment:
I'd be inclined to make `ApplyAliasActions` a static class (or even a `record`) and have the executor hold a reference to the enclosing `MetadataIndexAliasesService` instead.

Reply:
Very nice idea, more explicit than my approach. Thank you!

Reply:
Note that the executor still doesn't hold a reference to the enclosing service. If it did, you could effectively inline the `ApplyAliasActions#execute` method and then the task wouldn't need its `aliasesService` field.

(not sure if this change is still partly in flight, thought I'd say this just in case it isn't)

Reply:
Done.